Fix fakezod by tagging proper urbit/arvo version

s-ol opened the pull request:
urbit/arvo#master is not compatible with urbit 0.4.3, making fakezods impossible.

I also only cloned it in fakezod and removed the file you weren't using.

Layer count has been reduced by merging multiple RUN statemets where appropriate.
Urbit and APT are still seperate so the cache can work better when updating only a new urbit version.

s-ol committed:
cache apt and urbit seperately

s-ol committed:
cleanup with pinned arvo version

s-ol committed:
update readme

s-ol committed:
make fakezod a tag

paulbellamy commented on README.md:
```diff
@@ -35,17 +34,21 @@ $ docker run -ti -v `pwd`/fintud-macrep:/urbit/fintud-macrep paulbellamy/urbit f
 
-There is another image, with a different entrpoint to make this even
-easier:
+There is another image, with a different entrypoint and the arvo repo checekd
```
typo *checked

paulbellamy commented on README.md:
```diff
@@ -35,17 +34,21 @@ $ docker run -ti -v `pwd`/fintud-macrep:/urbit/fintud-macrep paulbellamy/urbit f
+
+to start it up after stopping it, omit `-A` and `-c`:
+
+    $ docker run -ti paulbellamy/urbit:fakezod piername
```
I'm assuming the change from a separate image to a separate tag will mean re-setting up the dockerhub automated builds? It's not a problem, just checking.

s-ol replied:
yes, you will need to add a second build-line on the original repo in docherhub and remove the other one.

paulbellamy reviewed (commented):
Thanks for the PR! Just a question and typo, then it will be ready to go.

s-ol commented:
>Better not merge yet, i have been tweaking things a bit and am not sure which state is currently on my master. I'll let you know when I think it is stable and you can do your own testing of course.

I think it's merge-ready now.

s-ol committed:
expose 8443

s-ol committed:
more extensive README

s-ol commented:
Looking better now. I need to test the fakezod and HTTP again, but you will notice i dropped the `useradd` stuff again - docker added a `--user` flag at some point and it was making the signal passing wonky anyway.

I also added an `EXPOSE 8443`, this allows linking the container to something like `nginx-proxy`, which will then autodetect the port with zero configuration.

Btw since you were asking, here is what I would set up the build settings like:
![2017-03-05-190958_880x370_scrot](https://cloud.githubusercontent.com/assets/1731279/23589951/59ee5bae-01d7-11e7-9663-33f1064d71d3.png)

s-ol committed:
add troubleshooting section to readme

paulbellamy commented:
Thanks very much!
